client/src/pages/preferences/PreferencesFieldSet.tsx

```diff
@@ -0,0 +1,311 @@
+import { gql } from "@apollo/client"
+import {
+  DEFAULT_PAGE_PROPS,
+  DEFAULT_SEARCH_PROPS,
+  SEARCH_OBJECT_TYPES
+} from "actions"
+import API from "api"
+import * as FieldHelper from "components/FieldHelper"
+import Fieldset from "components/Fieldset"
+import Messages from "components/Messages"
+import NavigationWarning from "components/NavigationWarning"
+import {
+  mapPageDispatchersToProps,
+  PageDispatchersPropType,
+  useBoilerplate,
+  usePageTitle
+} from "components/Page"
+import { FastField, Form, Formik } from "formik"
+import _get from "lodash/get"
+import React, { useMemo } from "react"
+import { Button } from "react-bootstrap"
+import { connect } from "react-redux"
+import Settings from "settings"
+import utils from "utils"
+import ExportFieldsPanel from "./ExportFieldsPanel"
+
+const GQL_GET_PREFERENCES = gql`
+  query ($preferenceQuery: PreferenceSearchQueryInput) {
+    preferenceList(query: $preferenceQuery) {
+      list {
+        uuid
+        name
+        type
+        category
+        description
+        defaultValue
+        allowedValues
+      }
+    }
+  }
+`
+interface UserPreference {
+  preference: { uuid: string }
+  value: string
+}
+
+interface PreferencesFieldsetProps {
+  pageDispatchers?: PageDispatchersPropType
+  category?: string
+  userPreferences?: UserPreference[]
+  onSubmit: (values: Record<string, any>, formikBag: any) => void
+  title?: string
+  actionLabel?: string
+  saveSuccess?: string | null
+  saveError?: any
+  exportObjectTypes?: string[]
+}
+
+function isExportFieldsPref(category: string) {
+  return category === "export"
+}
+
+function camelCaseToTitle(str: string) {
+  const withSpaces = str.replace(/([a-z])([A-Z])/g, "$1 $2")
+  return withSpaces.charAt(0).toUpperCase() + withSpaces.slice(1).toLowerCase()
+}
+
+function getLabelFromDictionary(preferenceName: string, field: string) {
+  const key = preferenceName.split("_")[0]
+  const label = Settings.fields[key]?.[field]?.label
+  return label ?? camelCaseToTitle(field)
+}
+
+function convertStringValueToType(value: any, type: string) {
+  if (type === "boolean") {
+    return String(value).toLowerCase() === "true"
+  }
+  return value
+}
+
+const PreferencesFieldset = ({
+  pageDispatchers,
+  category = null,
+  userPreferences = [],
+  onSubmit,
+  title = "Preferences",
+  actionLabel = "Save preferences",
+  saveSuccess,
+  saveError,
+  exportObjectTypes = []
+}: PreferencesFieldsetProps) => {
+  const { loading, error, data } = API.useApiQuery(GQL_GET_PREFERENCES, {
+    preferenceQuery: { category, pageSize: 0 }
+  })
+  const { done, result } = useBoilerplate({
+    loading,
+    error,
+    pageProps: DEFAULT_PAGE_PROPS,
+    searchProps: DEFAULT_SEARCH_PROPS,
+    pageDispatchers
+  })
+  usePageTitle(title)
```

Comment:
This still changes the page title when the search export dialog pops up. Probably best to not set the title here, but call `usePageTitle` from `Preferences.tsx` and `MyPreferences.tsx`.